cartridge/pool.lua

```diff
@@ -69,13 +83,82 @@ local function connect(uri, options)
         fiber.sleep(0)
     end
     vars.locks[uri] = true
-    local conn, err = e_connect:pcall(_connect, uri, options)
+    local conn, err = errors.pcall('NetboxConnectError', _connect, uri, options)
     vars.locks[uri] = false
 
     return conn, err
 end
 
+local function _gather_netbox_call(ret_map, uri, fn_name, args, opts)
+    local conn, err = connect(uri)
+    if conn == nil then
+        ret_map[uri] = {nil, err}
+        return
+    end
+
+    ret_map[uri] = {errors.netbox_call(conn, fn_name, args, opts)}
+    return
+end
+
+--- Perform a remote call to multiple URIs and map results.
+--
+-- (**Added** in v1.1.0-12)
+-- @function map_call
+-- @local
+--
+-- @tparam string fn_name
+-- @tparam ?table args
+-- @tparam[opt] table opts
+-- @tparam {string,...} opts.uri_list
+--   array of URIs for performing remote call
+-- @tparam ?number opts.timeout
+--   passed to `net.box` `conn:call` options
+--
+-- @treturn {URI=table,...}
+--   Call results mapping for every URI.
+--   Any errors are mapped as `[URI] = {nil, err}`
+local function map_call(fn_name, args, opts)
+    checks('string', '?table', {
+        uri_list = 'table',
+        timeout = '?', -- for net.box.call
+    })
+
+    local i = 0
+    for _, _ in pairs(opts.uri_list) do
+        i = i + 1
+        if type(opts.uri_list[i]) ~= 'string' then
+            error('bad argument opts.uri_list' ..
+                ' to ' .. (debug.getinfo(1, 'nl').name or 'map_call') ..
+                ' (contiguous array of strings expected)', 2
+            )
+        end
+    end
+
+    local ret_map = {}
+    local fibers = {}
+    for _, uri in pairs(opts.uri_list) do
+        local fiber = fiber.new(
+            _gather_netbox_call,
+            ret_map, uri, fn_name, args,
+            {timeout = opts.timeout}
+        )
+        fiber:name('netbox_call_map')
+        fiber:set_joinable(true)
+        fibers[uri] = fiber
+    end
+
+    for _, uri in pairs(opts.uri_list) do
```

Comment:
I'd suggest to add ell errors as second example - for simple cases when it invalidates the whole request. Otherwise user will always have to grind through result map searching for errors

Reply:
Is your proposal to return something like this?

```lua
  {3301 = ret3301, 3302 = nil},
  {3301 = nil, 3302 = err3302},
```

Reply:
Yes, something like this. I just want a quick way to understand if all was well